fenix/app/src/main/java/org/mozilla/fenix/utils/Settings.kt

```diff
@@ -1928,4 +1928,12 @@ class Settings(private val appContext: Context) : PreferencesHolder {
         default = { enableFxSuggest },
         featureFlag = FeatureFlags.fxSuggest,
     )
+
+    /**
+     * Indicates if the user is shown new redesigned Toolbar UI.
+     */
+    var enableRedesignedToolbar by booleanPreference(
+        key = appContext.getPreferenceKey(R.string.pref_key_toolbar_use_redesign),
+        default = FxNimbus.features.toolbar.value().toolbarRedesignEnabled,
```

Comment:
Should this be using `lazyFeatureFlagPreference` with a `featureFlag`? I am not fully sure if I am understanding shared preference myself, but from reading `lazyFeatureFlagPreference`, I believe while this is feature flagged off, then we will not be storing the `default` value into a user's shared preference and instead be using a dummy object that just returns false. 

If we used `booleanPreference` preference here, we will basically be setting `pref_key_toolbar_use_redesign` to be `false` by default for all users that has Fenix installed and receives this patch. Consequently, I think this will mean when we do enable this, we will have user with `pref_key_toolbar_use_redesign` already set to `false` since `toolbarRedesignEnabled` is false. I believe this is how it works, but I think I need someone to help me verify if what I am saying is actually true.

Reply:
GL is highlighting good points here. 

My 2 cents here would be to simply use nimbus flag directly via `RedesignToolbarFeature` and use the experiment settings to self enroll for users/testers who want to see this in action.

Reply:
> My 2 cents here would be to simply use nimbus flag directly via RedesignToolbarFeature and use the experiment settings to self enroll for users/testers who want to see this in action.

I would generally say we should have a secret setting pref that can enable the feature manually. Otherwise, you will also have to make a change to the nimbus flag during local development to enable the feature. We ran into the issue where we couldn't have UX review a design because they couldn't enable the feature themselves unless the nimbus experiment was running.

Reply:
hello!
great points all around, I'm actually going to rework this to remove the nimbus flag for now and check out `lazyFeatureFlagPreference`.
The way this feature will roll out will be to Nightly users for a bit and let it soak while we measure performance metrics and then eventually roll up to Beta if behavior changes are not negative for search revenue.

Reply:
@gabrielluong We could default nimbus flag to true only for development in the config though.

For my own curiosity, if we use `lazyFeatureFlagPreference`, the nimbus flag still needs to be true right? So either the nimbus flag has to be set to true for nightly config, or there should be a nightly experiment from the beginning that testers/UX can turn on in the experiment settings?

From the experiment testing perspective, I think the latter one seems better since it tests the experiment config in nightly as well, which would be the same as the beta experiment config, wdyt?

Reply:
> @gabrielluong We could default nimbus flag to true only for development in the config though.

I think it depends on the state of the feature development. Only when the feature is ready to ship in Nightly should the feature be enabled by default in Debug and Nightly. Enabling it by default for debug builds in Nimbus mean we are no longer testing the state that is shipped to our users, which will lead to other issues such as we are no longer catching problems early ourselves during day-to-day development, and we will also have to programmatically turn the nimbus flag on/off if we wanted to see the old/new state. We may run to issues where we even accidentally commit this temporary code change (this has already happened before with Compose Top Sites).

I see 2 hard requirements:
- We need a way to toggle on/off a feature via secret settings so that we don't have to programmatically make a change and rebuild every time we are doing trying to test the different feature states.
- We need a preference that can be used for Nimbus experimentation that could be override by the secret settings. 
To obtain both of these requirements, we do no have to solely depend on using the same preference key. It shouldn't also be that we can only do or the another. We need to be able to do both so that we aren't programmatically toggling the preference which will affect our developer experience.

> For my own curiosity, if we use lazyFeatureFlagPreference, the nimbus flag still needs to be true right? So either the nimbus flag has to be set to true for nightly config, or there should be a nightly experiment from the beginning that testers/UX can turn on in the experiment settings?

The Nimbus and feature flag would need to be true. If we set the preference too early without `lazyFeatureFlagPreference`, we are still running to the original problem where `booleanPreference` sets the value on the first read. I think there is possibly a deeper rooted problem with how we are using preferences to store the Nimbus default value. From the experiment testing perspective, we should only ever read from the Nimbus value getter directly instead of storing that value in a preference because once we create the shared preference with the Nimbus default value, we are kinda stuck with that default value that has been determined by the treatment group placed upon by experimenter. When we do let a feature ride to all the channels that has been using Nimbus and these preferences, we need to make sure their existing preferences are no longer used or removed completely. Otherwise, we might be run into a scenario where an user was in the treatment group where the feature was turned off, but the preference used here will never be reset unless we also remove the preference.